### PR TITLE
Remove deface reliance

### PR DESCRIPTION
**Pull request by jacobherrington**
Made this PR to the public repo to, but not sure it's perfect for that one. Should work for us.

### CODE REVIEW

File: config/initializers/braintree.rb [hunk 1]
```diff
@@ -1 +1,12 @@
 Spree::Admin::PaymentsController.helper :braintree_admin
+
+if Spree::Backend::Config.respond_to?(:menu_items) && Rails.env != "test"
```
Comment (fastjames) [inline config/initializers/braintree.rb]: I know you said that this was required for test, but I would expect to see a fully reciprocal conditional in the other change block (here, in `admin_navigation_menu.rb`) that forces the Deface code to run if you _are_ in test. Why is that not needed?

Comment (jacobherrington) [reply config/initializers/braintree.rb]: Good point. The issue is that the routes generated by the menu item in some cases rely on another extension and as such they don't exists and calling the Railsified `admin_#{whatever}_path` method blows up the app.

**Comment by fastjames** [comment]
@jacobherrington for our purposes I changed your PR to target the `gemini_master` branch. We can make a separate PR for upstream against their `master`.

**Comment by fastjames** [comment]
@jacobherrington Can you rebase this against `gemini_master` and resolve the conflicts? I think that will make it merge ready.

**Pull request merged**